src/content/docs/tutorial-external-services/solve-external.mdx

```diff
@@ -0,0 +1,89 @@
+
+Note that there is no one solution for problems with APIs and external services. Take the ideas in this tutorial as a jumping off point and dig in whenever you uncover something.
+
+## Objectives
+
+This doc uses `Sms notification`, the external service we identified [in the previous tutorial doc](docs/tutorial-external-services/respond-external), as an example of a problematic API or external service. This doc covers:
+
+* Looking for anomalies in metric charts
+* Using traces to find the root cause
+
+## Identify and solve the root cause
+
+You've already used the map view to identify the problematic external service, but we can use it to go even deeper:
+
+<Steps>
+    <Step>
+### Identify anomalies with metrics
+
+In the map view hover over the problematic API or external service and select **Show entity preview**. 
+
+<img
+    title="The preview pane for an external service"
+    alt="The preview pane for an external service"
+    src={apmPreview}
+    />  
+
+
+This opens up a pane where you get an overview of the general state of the service. You already used some of these metrics to identify it as the problem, but it's useful to take a deeper look here. Keep an eye out for:
+
+* Any spikes in the charts. 
+* Obviously high response rate or throughput.
+* If response rate and throughput hit highs and lows around the same time.
+
+If you identify an anomaly, use the time picker in the top right to expand the timerange you are looking at. This is useful to identify an anomaly as opposed to a pattern.
+
+In the screenshot above, nothing stands out except the throughput is a bit higher than you would have expected. Not anything to worry about immediately.
+    </Step>
+    <Step>
+### Find a code-level issue with traces         
+
+Hover over the problematic external service again and this time click **View traces**. 
+
+<img
+    title="The trace pane for an external service"
+    alt="The trace pane for an external service"
+    src={apmTrace}
+    />  
+
+This displays various charts about traces involved with that service. In this case, there is a single trace called `Controller/Sinatra//purchase` which allows you to follow the data created from your user clicking the buy button all the way to the API call that sends an SMS notification. 
+
+Click in as deep as you find useful, but focus again on finding anomalies or errors in the chart. For example, you might follow the trace all the way up to the API call that decrypts a user's stored phone number. An error here could uncover the whole problem. If the decrypt call fails and never returns a valid phone number, your user would never receive a phone notification. 
+
+This would also explain why you saw no errors in the `Sms notification` charts. That API was working correctly, but an API further upstream was giving it data that didn't actually serve a purpose.
```

Comment:
Is it worth explaining upstream vs. downstream? Also maybe I am missing something, but how do you know it's a problem upstream?